presto-verifier/src/main/java/com/facebook/presto/verifier/Validator.java

```diff
@@ -327,22 +328,33 @@ private QueryResult executeQueryTest()
             }
             while (retry < testTeardownRetries);
             // if teardown is not successful the query fails
-            queryResult = tearDownResult.getState() == State.SUCCESS ? queryResult : tearDownResult;
+            if (tearDownResult.getState() != State.SUCCESS) {
+                queryResult = tearDownResult;
+                if (tearDownResult.getException() != null) {
+                    if (setupResult.getException() != null) {
+                        tearDownResult.addSuppressed(setupResult.getException());
+                    }
+                    if (queryResult.getException() != null) {
+                        tearDownResult.addSuppressed(queryResult.getException());
+                    }
+                }
+            }
         }
         return queryResult;
     }
 
     private QueryResult executeQueryControl()
     {
         Query query = queryPair.getControl();
-        QueryResult queryResult = new QueryResult(State.INVALID, null, null, null, null, ImmutableList.of());
+        QueryResult setupResult = null;
+        QueryResult queryResult = null;
         try {
-            // startup
-            queryResult = setup(query, controlPreQueryResults, controlPrequery -> executeQuery(controlGateway, controlUsername, controlPassword, queryPair.getControl(), controlPrequery, controlTimeout, sessionProperties));
+            // setup
+            setupResult = setup(query, testPreQueryResults, testPrequery -> executeQuery(testGateway, testUsername, testPassword, queryPair.getTest(), testPrequery, testTimeout, sessionProperties));
```

Comment:
Why are we changing these to test? Are these typos?

Reply:
copy paste accident :)